### PR TITLE
feat(Events): allow a different local vs public name

### LINKED ISSUE #2272: The events property should allow renaming similar to properties
The following should work:

```
@Directive({
  events: ['renamedEvent1 : event1', 'event2']  
})
class MyDirective {
  renamedEvent1: EventEmitter = new EventEmitter();
  event2: EventEmiter = new EventEmitter();
}
```

### PR DESCRIPTION
**Pull request by vicb**
closes #2272

### CODE REVIEW

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 1]
```diff
@@ -380,8 +381,20 @@ export class BindingData {
   createEventEmitterAccessors() {
     if (!(this.binding instanceof DirectiveBinding)) return [];
     var db = <DirectiveBinding>this.binding;
-    return ListWrapper.map(db.eventEmitters, eventName => new EventEmitterAccessor(
-                                                 eventName, reflector.getter(eventName)));
+    return ListWrapper.map(db.eventEmitters, eventConfig => {
+      let localName;
```
Comment (vsavkin) [inline modules/angular2/src/core/compiler/element_injector.ts]: Maybe we should call them fieldName and eventName?

fieldName is not really local: Angular attaches a listener to it, developers can interact with it when loading components dynamically.

File: modules/angular2/test/core/compiler/integration_spec.ts [hunk 2]
```diff
@@ -694,6 +694,32 @@ export function main() {
                });
          }));
 
+      it('should support EventEmitter with local names',
```
Comment (vsavkin) [inline modules/angular2/test/core/compiler/integration_spec.ts]: I think we can leave only this test and remove the other one. This is a more general case, and we have unit tests checking the other case. 

Imo we have far too many integration tests at this point.

What do you think?

**Comment by vicb** [comment]
Thanks for the review, I have updated the PR, PTAL.

File: modules/angular2/test/core/compiler/element_injector_spec.ts [hunk 3]
```diff
@@ -418,6 +418,18 @@ export function main() {
         expect(accessor.getter(new HasEventEmitter())).toEqual('emitter');
       });
 
+      it('should allow a different internal vs public name', () => {
```
Comment (vsavkin) [inline modules/angular2/test/core/compiler/element_injector_spec.ts]: eventName and fieldName?

Comment (vicb) [inline modules/angular2/test/core/compiler/element_injector_spec.ts]: good catch, I'll update and merge

**Pull request closed without merging**